k2/csrc/cuda/tensor.cu

```diff
@@ -98,9 +111,9 @@ TensorPtr Tensor::Index(int32_t axis, int32_t index) const {
 }
 
 void Tensor::Init(ContextPtr c) {
-  int32_t nelement = shape_.Nelement();
+  int32_t storage_size = shape_.StorageSize();
```

Comment:
As `tensor` may be not contiguous, we need to use `StorageSize()` instead of `Nelement()` to allocate memory.